vmdb/app/models/policy_event/purging.rb

```diff
@@ -5,7 +5,7 @@ module Purging
 
     module ClassMethods
       def purge_date
-        (purge_config(:keep_policy_events) || 6.months).to_i_with_method.ago.utc
+        (purge_config(:keep_policy_events) || 6.months).to_i_with_method.seconds.ago.utc
```

Comment:
@matthewd 
The to_i_with_method does the conversion of the 6.months into a seconds and then does a to_i on it converting to Fixnum.

For duration we are expecting the user to enter data in different formats

:keep_policy_events = 6.months or "6.months" or "15552000" or 15552000

The to_i_with_method doesn't know the context of the data if it is a timeout/duration or a counter. So in the code once we read the data we have to convert it to seconds at the appropriate places. The to_i_with_method always returns a Fixnum which then has to be converted to ActiveSupport::Duration before we call ago on it